src/v/datalake/schema_protobuf.cc

```diff
@@ -120,22 +120,20 @@ field_outcome from_protobuf(
     /**
      * We support 32 bits unsigned integers by casting them to the iceberg long
      * type, the long type can hold the max value of uint32_t without an
-     * overflow. This isn't true for 64 bit unsigned integers and this is why we
-     * return a validation error in this case.
+     * overflow.
      * This the original behavior of Parquet as in:
      * https://github.com/apache/iceberg/blob/79fd977f67592a16579cff31478e7ea98ef126e4/parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java#L223
      *
+     * For a better experience for users using uint64_t types that are not
+     * normally supported in iceberg, we fallback to encoding them as strings.
+     *
      */
     case pb::FieldDescriptor::TYPE_UINT32:
     case pb::FieldDescriptor::TYPE_FIXED32:
         return success(fd, iceberg::long_type{});
     case pb::FieldDescriptor::TYPE_UINT64:
     case pb::FieldDescriptor::TYPE_FIXED64:
-        return schema_conversion_exception(fmt::format(
-          "Protocol buffer field {} type {} not supported, iceberg only "
-          "supports signed integers",
-          fd.DebugString(),
-          fd.type_name()));
+        return success(fd, iceberg::string_type{});
```

Comment:
At first I thought maybe we ought to use a fixed type, but thinking a bit more I think I'm on board since it's probably easier for query engines to cast a string

Maybe worth a discussion with product, though also something we can change if need be once we begin end-to-end testing and see how it plays with real query engines

Reply:
This was actually @mattschumpert's suggestion from reading the schema doc. I've already updated the doc there, so I think we're aligned unless we see something better from the query engines (but I strongly suspect it's more natural to parse numbers than bitwise cast them in these engines).

Reply:
Yep, makes total sense thanks!